core/arch/arm/kernel/boot.c

```diff
@@ -997,13 +997,17 @@ static int get_nsec_memory_helper(void *fdt, struct core_mmu_phys_mem *mem)
 	size_t n = 0;
 	int len = 0;
 
-	addr_size = fdt_address_cells(fdt, 0);
-	if (addr_size < 0)
-		return 0;
-
-	len_size = fdt_size_cells(fdt, 0);
-	if (len_size < 0)
-		return 0;
+	if (IS_ENABLED(_CFG_USE_DTB_OVERLAY)) {
+		len_size = sizeof(paddr_t) / sizeof(uint32_t);
+		addr_size = sizeof(paddr_t) / sizeof(uint32_t);
```

Comment:
When `_CFG_USE_DTB_OVERLAY` is enabled and the target memory already stores a DTB, OP-TEE is expected to update this DTB. In that case, size/address cell counts should still be read from the DTB image.